runelite-client/src/main/java/net/runelite/client/plugins/customcursor/CustomCursorPlugin.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2018, Kruithne <kruithne@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.customcursor;
+
+import com.google.inject.Provides;
+import java.awt.Cursor;
+import java.awt.Point;
+import java.awt.Toolkit;
+import javax.inject.Inject;
+import javax.swing.JPanel;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.ClientUI;
+
+@PluginDescriptor(
+	name = "Custom Cursor",
+	description = "Configuration for the Custom Cursor plug-in",
+	enabledByDefault = false
+)
+public class CustomCursorPlugin extends Plugin
+{
+	@Inject
+	private ClientUI clientUI;
+
+	@Inject
+	private CustomCursorConfig config;
+
+	@Provides
+	CustomCursorConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(CustomCursorConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		updateCursor();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		clientUI.getContainer().setCursor(Cursor.getDefaultCursor());
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("customcursor") && event.getKey().equals("cursorStyle"))
+		{
+			updateCursor();
```

Comment:
This can run off the swing thread, does that matter?

Reply:
As far as I can tell it does not. I ran it on the client thread and it worked fine. I also checked the docs. It is possible i'm wrong though because I know little about multithreading.